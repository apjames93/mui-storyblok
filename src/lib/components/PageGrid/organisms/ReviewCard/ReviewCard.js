import React, { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import { useFadedShadowStyles } from '@mui-treasury/styles/shadow/faded';
import { useWideCardMediaStyles } from '@mui-treasury/styles/cardMedia/wide';
import { renderComponentsWithBridge } from 'lib/utils/customComponents';
import TextInfo from 'lib/components/PageGrid/atoms/TextInfo/TextInfo';
import CardMedia from 'lib/components/PageGrid/atoms/CardMedia/CardMedia';

const CardActions = lazy(() => import('lib/components/PageGrid/molecules/CardActions/CardActions'));

const useStyles = makeStyles(() => ({
  root: {
    overflow: 'initial',
    maxWidth: 304,
    backgroundColor: 'transparent',
  },
  title: {
    marginBottom: 0,
  },
  rateValue: {
    fontWeight: 'bold',
    marginTop: 2,
  },
  content: {
    position: 'relative',
    padding: 24,
    margin: '-24% 16px 0',
    backgroundColor: '#fff',
    borderRadius: 4,
  },
  favorite: {
    position: 'absolute',
    top: 12,
    right: 12,
  },
  locationIcon: {
    marginRight: 4,
    fontSize: 18,
  },
}));

export const ReviewCard = React.memo(({
  image,
  overline,
  heading,
  body,
  imgAlt,
  imgTitle,
  dataBlokC,
  dataBlokUid,
  storyblokClass,
  cardActions,
}) => {
  const styles = useStyles();
  const mediaStyles = useWideCardMediaStyles();
  const shadowStyles = useFadedShadowStyles();
  const actions = cardActions[0];

  return (
    <Card
      className={cx(styles.root, storyblokClass)}
      data-blok-c={dataBlokC}
      data-blok-uid={dataBlokUid}
    >
      <CardMedia
        image={image}
        style={{ ...styles.media, ...mediaStyles.root }}
        alt={imgAlt}
        title={imgTitle}
      />
      <CardContent
        className={cx(shadowStyles.root, styles.content)}
      >
        <TextInfo
          overline={overline}
          heading={heading}
          body={body}
        />
        {actions ? (
          <Suspense fallback={<></>}>
            {renderComponentsWithBridge({ CardActions }, actions)}
          </Suspense>
        ) : null}
      </CardContent>
    </Card>
  );
});

export default ReviewCard;

ReviewCard.propTypes = {
  /** actions to add in card */
  cardActions: PropTypes.arrayOf(PropTypes.shape()),
  /** overline heading  */
  heading: PropTypes.string.isRequired,
  /** overline body  */
  body: PropTypes.string.isRequired,
  /** alt text for the image if url cant display */
  imgAlt: PropTypes.string.isRequired,
  /** the title for the image */
  imgTitle: PropTypes.string.isRequired,
  /**
   * url for image
   * Image to be displayed as a background image
   * */
  image: PropTypes.string.isRequired,
  /** overline text */
  overline: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokC: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokUid: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  storyblokClass: PropTypes.string,
};


ReviewCard.defaultProps = {
  cardActions: [],
  dataBlokC: '',
  dataBlokUid: '',
  storyblokClass: '',
  overline: '',
};
