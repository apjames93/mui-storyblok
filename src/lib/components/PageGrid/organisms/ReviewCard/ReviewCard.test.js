import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ReviewCard from './ReviewCard';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "ReviewCard", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    overline: 'overline',
    heading: 'heading',
    body: 'body',
    image: 'cool.jpg',
    imgAlt: 'cool',
    imgTitle: 'imgTitle',
    icon: [{
      component: 'Icon',
      iconName: 'star',
    }],
    cardActions: [{
      _editable: '<!--#storyblok#{"name": "CardActions", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'CardActions',
      content: [{
        _editable: '<!--#storyblok#{"name": "ButtonRedirect", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'ButtonRedirect',
        redirectRoute: 'page-welcome',
        button: [{
          _editable: '<!--#storyblok#{"name": "Button", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
          component: 'Button',
          buttonText: 'buttonText',
        }],
      }],
    }],
  };
  const comp = shallow(<ReviewCard {...props} />);
  return { comp, props };
}

describe('<ReviewCard />', () => {
  it('renders ReviewCard', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<ReviewCard {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
