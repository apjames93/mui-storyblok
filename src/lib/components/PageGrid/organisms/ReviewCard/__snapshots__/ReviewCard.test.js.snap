// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCard /> snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-blok-c=""
  data-blok-uid=""
>
  <div
    alt="cool"
    className="MuiCardMedia-root makeStyles-root-12 "
    data-blok-c=""
    data-blok-uid=""
    style={
      Object {
        "backgroundImage": "url(\\"cool.jpg\\")",
      }
    }
    title="imgTitle"
  />
  <div
    className="MuiCardContent-root makeStyles-root-9 makeStyles-content-4"
  >
    <span
      className="MuiTypography-root TextInfoContent-overline-13 MuiTypography-body1"
    >
      overline
    </span>
    <h4
      className="MuiTypography-root TextInfoContent-heading-14 MuiTypography-body1"
    >
      heading
    </h4>
    <p
      className="MuiTypography-root TextInfoContent-body-15 MuiTypography-body1"
    >
      body
    </p>
  </div>
</div>
`;
